lib/utils/helpers.ts

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright 2016-2017 Resin.io
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import os = require('os');
+import Promise = require('bluebird');
+import _ = require('lodash');
```

Comment:
The general approach is that we put require calls inside methods _for files that have to be loaded at startup time_, not for all files everywhere. Almost all the utils files etc just use the traditional top-level `require`, it's just the CLI setup code & the actions themselves that delay requiring modules. It's safe-ish because the utils modules like this one get loaded async in the first place, only once we're running the action that needs them, so we know they're required regardless (and each of these files is similar enough that it's not worth splitting loading further).

All of this is a bit messy though, you're totally right. I'd love to do it better, but I think the main fix for this might really be going further down the bundling & packaging route, so all requires are drastically faster in the first place, and we can simplify the code completely. Might also be some webpack code splitting magic available, and I think Etcher are looking into some of that as we speak. We'll see...

Reply:
Thanks for the detailed insights :+1: